instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integration/rum/EndToEndRumFragmentTrackingTests.kt

```diff
@@ -39,84 +39,131 @@ internal class EndToEndRumFragmentTrackingTests {
+                        time.asLong >= 0
+                    }
+                )
             )
         )
+
         // swipe to change the fragment
         onView(withId(R.id.tab_layout)).perform(swipeLeft())
         instrumentation.waitForIdleSync()
         Thread.sleep(200) // give time to the view id to update
         val fragmentBViewUrl = currentFragmentViewUrl()
         mockServerRule.activity.supportFragmentManager.fragments
+        // for updating the time
         expectedEvents.add(
             ExpectedViewEvent(
                 fragmentBViewUrl,
                 2,
-                currentFragmentExtras()
+                currentFragmentExtras(),
+                extraViewAttributes = mapOf(
+                    "loading_type" to "fragment_display"
+                ),
+                extraViewAttributesWithPredicate = mapOf(
+                    "loading_time" to { time ->
+                        time.asLong >= 0
+                    }
+                )
             )
         )
-        // for updating the time
+        // view stopped
         expectedEvents.add(
             ExpectedViewEvent(
                 fragmentBViewUrl,
                 3,
-                currentFragmentExtras()
+                currentFragmentExtras(),
+                extraViewAttributes = mapOf(
+                    "loading_type" to "fragment_display"
+                ),
+                extraViewAttributesWithPredicate = mapOf(
+                    "loading_time" to { time ->
+                        time.asLong >= 0
+                    }
+                )
             )
         )
 
         // swipe to close the view
-        onView(withId(R.id.tab_layout)).perform(swipeLeft())
+        onView(withId(R.id.tab_layout)).perform(swipeRight())
         instrumentation.waitForIdleSync()
         Thread.sleep(200) // give time to the view id to update
 
         val fragmentCViewUrl = currentFragmentViewUrl()
+        // for updating the time
         expectedEvents.add(
             ExpectedViewEvent(
-                fragmentCViewUrl,
+                fragmentAViewUrl,
                 2,
-                currentFragmentExtras()
+                currentFragmentExtras(),
+                extraViewAttributes = mapOf(
+                    "loading_type" to "fragment_redisplay"
+                ),
+                extraViewAttributesWithPredicate = mapOf(
+                    "loading_time" to { time ->
+                        time.asLong >= 0
+                    }
+                )
             )
         )
-        // for updating the time
+
+        // view stopped
         expectedEvents.add(
             ExpectedViewEvent(
-                fragmentCViewUrl,
+                fragmentAViewUrl,
                 3,
-                currentFragmentExtras()
+                currentFragmentExtras(),
+                extraViewAttributes = mapOf(
+                    "loading_type" to "fragment_redisplay"
```

Comment:
how does that work?

Reply:
We are instrumenting the test application to perform swipes between the fragments. We are intercepting the events as you are doing with your Python mock server and we are asserting them at the end. In this case this fragment (view) was already displayed once so when we detect that and send `fragment_redisplay` for the `loading_type`